component/src/main/java/io/siddhi/extension/store/mongodb/MongoDBEventTable.java

```diff
@@ -623,4 +626,262 @@ protected void destroy() {
+        compiledGroupByJSON.append((groupByAttributesList.size() == 1) ? ',' : "},");
+        List<MongoSelectExpressionVisitor> getSelectExpressionVisitorList =
+                getSelectAttributesList(selectAttributeBuilders);
+        for (int i = 0; i < getSelectExpressionVisitorList.size(); i++) {
+            MongoSelectExpressionVisitor visitor = getSelectExpressionVisitorList.get(i);
+            String rename = selectAttributeBuilders.get(i).getRename();
+            if (!groupByAttributesList.contains(rename)) {
+                if (visitor.getStreamVarCount() == 0 && visitor.getConstantCount() == 0) {
+                    String compiledCondition = visitor.getCompiledCondition();
+                    boolean isFunctionUsed = visitor.checkFunctionsUsed();
+                    if (isFunctionUsed) {
+                        compiledGroupByJSON.append(rename).append(":").append(compiledCondition);
+                    } else {
+                        compiledGroupByJSON.append(rename).append(":{$last:").append(compiledCondition)
+                                .append('}');
+                    }
+                    compiledGroupByJSON.append(',');
+                }
+            }
+            if (getSelectExpressionVisitorList.indexOf(visitor) == (getSelectExpressionVisitorList.size() - 1)) {
+                compiledGroupByJSON.append(groupBySelectString).append("}");
+            }
+        }
+        compiledGroupByJSON.append("}");
+        return compiledGroupByJSON.toString();
+    }
+
+    private String getGroupByProjectionString(List<SelectAttributeBuilder> selectAttributeBuilders,
+                                              List<ExpressionBuilder> groupByExpressionBuilders) {
+        StringBuilder compiledProjectionJSON = new StringBuilder();
+        List<MongoSelectExpressionVisitor> getSelectExpressionVisitorList =
+                getSelectAttributesList(selectAttributeBuilders);
+        compiledProjectionJSON.append("{$project:{_id:0,");
+        for (int i = 0; i < getSelectExpressionVisitorList.size(); i++) {
+            MongoSelectExpressionVisitor visitor = getSelectExpressionVisitorList.get(i);
+            String rename = selectAttributeBuilders.get(i).getRename();
+            compiledProjectionJSON.append(rename).append(":1");
+            if (getSelectExpressionVisitorList.indexOf(visitor) == (getSelectExpressionVisitorList.size() - 1)) {
+                compiledProjectionJSON.append("}");
+            } else {
+                compiledProjectionJSON.append(",");
+            }
+        }
+        compiledProjectionJSON.append("}");
+        return compiledProjectionJSON.toString();
+    }
+
+    private String getHavingString(ExpressionBuilder havingExpressionBuilder) {
+        MongoExpressionVisitor visitor = new MongoExpressionVisitor();
+        havingExpressionBuilder.build(visitor);
+        String having = visitor.getHavingCompiledCondition();
+        return "{$match:" + having + "}";
```

Comment:
No need for having variable
```suggestion
        return "{$match:" + visitor.getHavingCompiledCondition() + "}";
```

Reply:
Fixed in
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/94fba768238670966474ed6ac8a757455790895b